#163601729 get all offices endpoint

fantastic-genius opened the pull request:
#### What does this PR do?
Implements get all offices endpoint
#### Description of Task to be completed?
 - implements function to get  office in officeController
- implements route for getting all office endpoint
- implements test for the get offices endpoint in officeApiTest
-refactor how id is generated when party is created in partyController
#### How should this be manually tested?
BASE_URL/api/v1/offices
#### Any background context you want to provide?
None
#### Screenshots (if appropriate)
None

fantastic-genius committed:
feat(partyController, partyMiddleware, partyRoutes, partiesApiTest): implement edit a party endpoint
- added edit party function in the controller
- validate the parameters passed in partyMiddleware
- created the edit party route in partyRoutes
[Delivers #163519782]

fantastic-genius committed:
merged ft-get-a-specific-party-endpoint-163519260 branch to develop

fantastic-genius committed:
Merge branch 'develop' of https://github.com/fantastic-genius/politico into develop

fantastic-genius committed:
Merge branch 'develop' into ft-edit-a-party-endpoint-163519782

fantastic-genius committed:
fix made correction to get a party endpoint

fantastic-genius committed:
feat(partyController, partyMiddleware, partyRoutes, partiesApiTest): implement delete party endpoint
- created delete party function in partyController
- implemented deleteparty parameters validation in partyMiddleware
- create delete party endpoint test in partiesApiTest
- created route for delete party endpoint
[Delivers #163519264]

fantastic-genius committed:
feat(partyController, db, partyMiddleware, partiesApiTest, officeController, officeMiddleware, officeRoute, officeApiTest): create an offfice endpoint
- implement function to create an office officeController
- implement function to validate client data for creation of officee in officeMiddleware
- implement route for create office endpoint
[Delivers #163601705]

fantastic-genius committed:
feat(partyController,  officeController, officeRoute, officeApiTest): get all offfice endpoint
- implements function to get  office in officeController
- implements route for getting all office endpoint
- implements test for the get offices endpoint in officeApiTest
-refactor how id is generated when party is created in partyController
[Delivers #163601729)]

veeqtor commented on server/controller/partyController.js:
```diff
@@ -69,10 +77,50 @@ class PartyController{
 
         return res.status(400).send({
```
Try and read up on HTTP statues, so you can know how and when to use the appropriate status code.

fantastic-genius replied:
Ok. Noted thanks

veeqtor commented on server/controller/partyController.js:
```diff
@@ -69,10 +77,50 @@ class PartyController{
         
     }
+
+    editAParty(req, res){
```
Nice work, You will be making a PATCH request to this endpoint, therefore the values to update the party with will be gotten from the JSON body and not the URL params, Please make the necessary changes for this implementation.

fantastic-genius replied:
ok,i would refactor my code

veeqtor commented on server/controller/partyController.js:
```diff
@@ -69,10 +77,50 @@ class PartyController{
+        
+    }
+
+    deleteParty(req, res){
```
What happens when I try to delete a non-existing party?

fantastic-genius replied:
Ok i will take that into consideration. Thanks

fantastic-genius committed:
Fix made followig changes
- implement response for when party to delete doesn't exist
- implements getting response from json data
- added correct status code for party not found

veeqtor commented on server/controller/partyController.js:
```diff
@@ -75,20 +75,21 @@ class PartyController{
     editAParty(req, res){
-        const id = parseInt(req.params.id);
-        const name = req.params.name;
+        const param_id = parseInt(req.params.id);
```
You are still getting the value from the URL params which is not supposed to be so., Please get your details from the JSON body.

fantastic-genius replied:
Ok

fantastic-genius committed:
Fix made followig changes
- implements getting response from json data

veeqtor commented on server/routes/partyRoutes.js:
```diff
@@ -8,7 +8,7 @@ const partyVersionedAPi = "/api/v1/parties/"
 router.get(partyVersionedAPi, partyController.getAllParty)
 router.get(`${partyVersionedAPi}:id`, partyController.getAParty)
-router.patch(`${partyVersionedAPi}:id/:name`, partyMiddleware.editPartyMiddleware, partyController.editAParty)
+router.patch(`${partyVersionedAPi}:id`, partyMiddleware.editPartyMiddleware, partyController.editAParty)
```
I believe this route should be in this format: `/parties/<party-id>/name`
